src/tdastro/sources/agn.py

```diff
@@ -0,0 +1,420 @@
+        blackhole_mass : float, optional
+            The black hole mass in g.
+            Default: 1e9 * M_SUN_G
+
+        Returns
+        -------
+        tau_v : np.ndarray
+            A length W array with the timescale in s for each wavelength.
+        """
+        # Equation and parameters for A=2.4, B=0.17, C=0.03, and D=0.21 adopted
+        # from Suberlak et al. 2021: DOI 10.3847/1538-4357/abc698
+        return 10 ** (
+            2.4
+            + 0.17 * np.log10(wavelength / (4000e-8))
+            + 0.03 * (mag_i + 23)
+            + 0.21 * np.log10(blackhole_mass / (1e9 * M_SUN_G))
+        )
+
+    @staticmethod
+    def compute_temp_at_r_0(M, Mdot, r_in):
+        """Compute the effective temperature at r0. This is the same as the maximum effective
+        temperature at the disc surface (Tmax).
+
+        Parameters
+        ----------
+        M : float
+            The mass of the gravitating centre.
+        Mdot : float
+            The accretion rate at the previous time step.
+        r_in : float
+            The inner radius of the accretion disk.
+
+        Returns
+        -------
+        T_0 : float
+            The effective temperature at r0.
+        """
+        # Lipunova, G., Malanchev, K., Shakura, N. (2018). Page 33 for the main equation
+        # DOI https://doi.org/10.1007/978-3-319-93009-1_1
+        sigma_sb = constants.sigma_sb.cgs.value
+        G = constants.G.cgs.value
+        return 2 ** (3 / 4) * (3 / 7) ** (7 / 4) * (G * M * Mdot / (np.pi * sigma_sb * r_in**3)) ** (1 / 4)
+
+    def compute_flux(self, times, wavelengths, graph_state, **kwargs):
+        """Draw effect-free observations for this object.
+
+        Parameters
+        ----------
+        times : `numpy.ndarray`
+            A length T array of rest frame timestamps.
+        wavelengths : `numpy.ndarray`, optional
+            A length N array of wavelengths (in angstroms).
+        graph_state : `GraphState`
+            An object mapping graph parameters to their values.
+        **kwargs : `dict`, optional
+            Any additional keyword arguments.
+
+        Returns
+        -------
+        flux_density : `numpy.ndarray`
+            A length T x N matrix of SED values (in nJy).
+        """
+        params = self.get_local_params(graph_state)
+
+        # Compute the parameters for these wavelengths.
+        tau_v = self.compute_tau_v_drw(wavelengths, params["mag_i"], params["blackhole_mass"])
+        sf_inf = self.compute_structure_function_at_inf(
+            wavelengths, params["mag_i"], params["blackhole_mass"]
+        )
+
+        # Compute the average flux of a standard disk model. Use a factor of 2 (two sides
+        # of the disk) to get the total flux.
+        fnu_average = 2.0 * self.compute_flux_standard_disk(
+            params["blackhole_accretion_rate"],
+            constants.c.cgs.value / wavelengths,  # nu
+            1,  # rin
+            0,  # i
+            10.0 * PARSEC_TO_CM,  # d
```

Comment:
I think we should change it to 1.0, because SNANA uses weird SED units, while we use CGS.
```suggestion
            1.0,  # d
```